vsts/pipelines/templates/_releaseTemplate.yml

```diff
@@ -0,0 +1,120 @@
+parameters:
+  ascName: OryxMCR
+  acrName: oryxdevmcr.azurecr.io
+  acrProdName: oryxmcr.azurecr.io
+
+steps:
+
+- checkout: self
+  clean: true
+
+- script: |
+    if [ "$(ReleaseBuildImages)" != "true" ] && [ "$(ReleaseRuntimeImages)" != "true" ]
+    then
+      echo "Invalid configuration."
+      echo "Variable 'ReleaseBuildImages' or 'ReleaseRuntimeImages' needs to be 'true' to run this"
+      exit 1
+    fi
+  displayName: 'Validate release pipeline run'
+
+- task: Docker@1
+  displayName: Container registry login
+  inputs:
+    command: login
+    azureSubscriptionEndpoint: ${{ parameters.ascName }}
+    azureContainerRegistry: ${{ parameters.acrName }}
+
+- task: DownloadBuildArtifacts@0
+  displayName: 'Download Build Artifacts'
+  inputs:
+    buildType: specific
+    project: '0bdbc590-a062-4c3f-b0f6-9383f67865ee'
+    pipeline: 11021
+    specificBuildWithTriggering: true
+    artifactName: drop
+
+- task: CopyFiles@2
+  displayName: 'Copy Files to: $(Build.ArtifactStagingDirectory)'
+  inputs:
+    SourceFolder: '$(System.ArtifactsDirectory)'
+    TargetFolder: '$(Build.ArtifactStagingDirectory)'
+
+- task: ms-devlabs.utilitytasks.task-Shellpp.Shell++@0
+  displayName: 'Pull and create release tags for build images'
+  inputs:
+    type: FilePath
+    scriptPath: ./vsts/scripts/create-releasetags-buildimage.sh
+  condition: and(succeeded(), eq(variables['ReleaseBuildImages'], 'true'))
+
+- task: ms-devlabs.utilitytasks.task-Shellpp.Shell++@0
+  displayName: 'Pull and create release tags for runtime images'
+  inputs:
+    type: FilePath
+    scriptPath: ./vsts/scripts/create-releasetags-runtimeimages.sh
+  condition: and(succeeded(), eq(variables['ReleaseRuntimeImages'], 'true'))
+
+- task: Docker@0
+  displayName: 'Push Build image to DockerHub'
```

Comment:
Are these `push` tasks correct? Here you are supplying a image names path file...where is this file coming from? in the earlier step we create tags, why can't we directly call `docker push` on those tags itself (and within that same script)

Reply:
yepp its correct, pushing images from build-images-mcr.txt which I create in the release scripts